libr/bin/format/pe/pe.c

```diff
@@ -765,10 +768,30 @@ static int bin_pe_init_hdr(RBinPEObj* pe) {
 		" e_magic e_cblp e_cp e_crlc e_cparhdr e_minalloc e_maxalloc"
 		" e_ss e_sp e_csum e_ip e_cs e_lfarlc e_ovno e_res e_oemid"
 		" e_oeminfo e_res2 e_lfanew", 0);
+	
+	// Enhanced validation for e_lfanew
 	if (pe->dos_header->e_lfanew > (unsigned int) pe->size) {
-		pe_printf ("Invalid e_lfanew field\n");
+		pe_printf ("Invalid e_lfanew field: 0x%x exceeds file size 0x%x\n", 
+			pe->dos_header->e_lfanew, pe->size);
+		return false;
+	}
+	
+	// Additional validation: e_lfanew should be reasonable (typically > 0x40)
+	if (pe->dos_header->e_lfanew < 0x40) {
+		pe_printf ("Suspicious e_lfanew field: 0x%x (too small)\n", pe->dos_header->e_lfanew);
+	}
+	
+	// Check alignment - NT headers should be aligned
+	if (pe->dos_header->e_lfanew & 3) {
+		pe_printf ("Warning: e_lfanew not aligned: 0x%x\n", pe->dos_header->e_lfanew);
+	}
+	
+	// Ensure we have enough space for NT headers
+	if (pe->dos_header->e_lfanew + sizeof(PE_(image_nt_headers)) > pe->size) {
+		pe_printf ("Invalid e_lfanew: NT headers would exceed file size\n");
```

Comment:
R_LOG_WARN too. and do not return false. we should go on and try out best to map the bin